diff --git NuttX/nuttx/include/ctype.h NuttX/nuttx/include/ctype.h
index fdcaad8..26550f8 100644
--- NuttX/nuttx/include/ctype.h
+++ NuttX/nuttx/include/ctype.h
@@ -49,6 +49,24 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* N.B. The following definitions are enabled at this time to allow the PX4
+ * development to continue until there is a SAFE  solution to foreign
+ * (non-nuttx) header file inclusion. There is a potential of a binary
+ * incompatibility and runtime errors, memory overwrites or corruption
+ * VVVVVVVVVVVVVVVVVVVVVVVVVV Begin Warning VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
+ */
+
+#define _U  01
+#define _L  02
+#define _N  04
+#define _S  010
+#define _P  020
+#define _C  040
+#define _X  0100
+#define _B  0200
+
+/*^^^^^^^^^^^^^^^^^^^^^^^^^ End Warning ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^*/
+
 /****************************************************************************
  * Name: isspace
  *
