diff --git NuttX/nuttx/include/nuttx/fs/ioctl.h NuttX/nuttx/include/nuttx/fs/ioctl.h
index e404057..7db7c91 100644
--- NuttX/nuttx/include/nuttx/fs/ioctl.h
+++ NuttX/nuttx/include/nuttx/fs/ioctl.h
@@ -370,7 +370,7 @@
 #define _LOOPIOC(nr)      _IOC(_LOOPBASE,nr)
 
 /* Modem driver ioctl definitions ********************************************/
-/* see nuttx/include/modem/*.h */
+/* see nuttx/include/modem/\*.h */
 
 #define _MODEMIOCVALID(c)  (_IOC_TYPE(c)==_MODEMBASE)
 #define _MODEMIOC(nr)      _IOC(_MODEMBASE,nr)
