diff --git NuttX/apps/platform/Makefile NuttX/apps/platform/Makefile
index 0e811ff..ccbd221 100644
--- NuttX/apps/platform/Makefile
+++ NuttX/apps/platform/Makefile
@@ -116,8 +116,7 @@ endif
 
 # Build context setup
 
-$(PLATFORMDIR): $(TOPDIR)$(DELIM).config
-	@echo "LN: platform$(DELIM)board to $(LINKDIR)"
+$(PLATFORMDIR):
 	$(Q) $(DIRUNLINK) $(PLATFORMDIR)
 	$(Q) $(DIRLINK) $(LINKDIR) $(PLATFORMDIR)
 
