diff --git NuttX/apps/Makefile NuttX/apps/Makefile
index 1b580c0..7ba85ff 100644
--- NuttX/apps/Makefile
+++ NuttX/apps/Makefile
@@ -79,13 +79,13 @@ all: $(BIN)
 .PHONY: import install context context_serialize context_rest .depdirs preconfig depend clean distclean
 
 define MAKE_template
-	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
+	+$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
 
 endef
 
 define SDIR_template
 $(1)_$(2):
-	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
+	+$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
 
 endef
 
@@ -108,13 +108,13 @@ install: $(BIN_DIR) .install
 .import: $(BIN) install
 
 import:
-	$(Q) $(MAKE) .import TOPDIR="$(APPDIR)$(DELIM)import"
+	+$(Q) $(MAKE) .import TOPDIR="$(APPDIR)$(DELIM)import"
 
 context_rest: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_context)
 
 context_serialize:
-	$(Q) $(MAKE) -C builtin context TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"
-	$(Q) $(MAKE) context_rest
+	+$(Q) $(MAKE) -C builtin context TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"
+	+$(Q) $(MAKE) context_rest
 
 context: context_serialize
 
