diff --git a/arch/arm/src/stm32/stm32_allocateheap.c b/arch/arm/src/stm32/stm32_allocateheap.c
index a222253..c77f81b 100644
--- a/arch/arm/src/stm32/stm32_allocateheap.c
+++ b/arch/arm/src/stm32/stm32_allocateheap.c
@@ -295,10 +295,13 @@
 #    define SRAM1_END 0x20018000
 #  elif defined(CONFIG_STM32_STM32F427) || defined(CONFIG_STM32_STM32F429)
 #    define SRAM1_END 0x20030000
+#  elif defined(CONFIG_STM32_STM32F446)
+#    define SRAM1_END 0x20020000
+#  elif defined(CONFIG_STM32_STM32F469)
+#    define SRAM1_END 0x20050000
 #  else
 #    define SRAM1_END 0x20020000
 #  endif
-
    /* Set the range of CCM SRAM as well (although we may not use it) */
 
 #  define SRAM2_START 0x10000000
