diff --git NuttX/nuttx/arch/arm/src/stm32f7/stm32_serial.c NuttX/nuttx/arch/arm/src/stm32f7/stm32_serial.c
index 5b57549..2c1c0ac 100644
--- NuttX/nuttx/arch/arm/src/stm32f7/stm32_serial.c
+++ NuttX/nuttx/arch/arm/src/stm32f7/stm32_serial.c
@@ -290,7 +290,7 @@ struct up_dev_s
   bool              rxenable;  /* DMA-based reception en/disable */
   uint16_t          rxdmain;   /* Next byte in the DMA where hardware will write */
   uint16_t          rxdmaout;  /* Next byte in the DMA buffer to be read */
-  VAR char * const  rxfifo;    /* Receive DMA buffer */
+  char       *const rxfifo;    /* Receive DMA buffer */
 #endif
 
 #ifdef HAVE_RS485
