diff --git NuttX/nuttx/include/cxx/clocale NuttX/nuttx/include/cxx/clocale
index 39d691e..154bf80 100644
--- NuttX/nuttx/include/cxx/clocale
+++ NuttX/nuttx/include/cxx/clocale
@@ -46,11 +46,10 @@
 // Namespace
 //***************************************************************************
 
-#ifdef CONFIG_LIBC_LOCALE
 namespace std
 {
   using ::setlocale;
+  using ::localeconv;
 }
-#endif
 
 #endif // __INCLUDE_CXX_CLOCALE
diff --git NuttX/nuttx/include/locale.h NuttX/nuttx/include/locale.h
index 501990d..16923e4 100644
--- NuttX/nuttx/include/locale.h
+++ NuttX/nuttx/include/locale.h
@@ -43,8 +43,6 @@
 #include <nuttx/config.h>
 #include <nuttx/compiler.h>
 
-#ifdef CONFIG_LIBC_LOCALE
-
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
@@ -109,5 +107,4 @@ FAR struct lconv *localeconv(void);
 }
 #endif
 
-#endif /* CONFIG_LIBC_LOCALE */
 #endif /* __INCLUDE_LOCALE_H */
diff --git NuttX/nuttx/libc/locale/lib_setlocale.c NuttX/nuttx/libc/locale/lib_setlocale.c
index 335eda8..fbd621a 100644
--- NuttX/nuttx/libc/locale/lib_setlocale.c
+++ NuttX/nuttx/libc/locale/lib_setlocale.c
@@ -43,6 +43,7 @@
 #include <locale.h>
 
 #ifdef CONFIG_LIBC_LOCALE
+#include <string.h>
 
 /****************************************************************************
  * Public Functions
@@ -62,8 +63,8 @@
 
 FAR char *setlocale(int category, FAR const char *locale)
 {
-  /* NULL indicates the the locale was not changed */
 
-  return NULL;
+  return ((strcmp (locale, "POSIX") == 0 || strcmp (locale, "C") == 0 ||
+           strcmp (locale, "") == 0) ? "C" :  NULL);
 }
 #endif
