diff --git NuttX/apps/Makefile NuttX/apps/Makefile
index 76b23ed..6d5462a 100644
--- NuttX/apps/Makefile
+++ NuttX/apps/Makefile
@@ -87,7 +87,7 @@ all: $(BIN)
 
 define SDIR_template
 $(1)_$(2):
-	$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
+	+$(Q) $(MAKE) -C $(1) $(2) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" BIN_DIR="$(BIN_DIR)"
 endef
 
 $(foreach SDIR, $(CONFIGURED_APPS), $(eval $(call SDIR_template,$(SDIR),all)))

diff --git NuttX/apps/examples/elf/Makefile NuttX/apps/examples/elf/Makefile
index c813e1e..60b059e 100644
--- NuttX/apps/examples/elf/Makefile
+++ NuttX/apps/examples/elf/Makefile
@@ -98,7 +98,9 @@ really_build: $(OBJS)
 	@touch .built
 
 .built:
+ifeq ($(CONFIG_EXAMPLES_ELF),y)
 	@$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
+endif
 	@$(MAKE) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" really_build
 
 ifeq ($(CONFIG_BUILD_KERNEL),y)
@@ -125,7 +127,9 @@ context:
 depend: .depend
 
 clean_tests:
+ifeq ($(CONFIG_EXAMPLES_ELF),y)
 	@$(MAKE) -C tests TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV) clean
+endif
 
 clean: clean_tests
 	$(call DELFILE, .built)
diff --git NuttX/apps/examples/module/Makefile NuttX/apps/examples/module/Makefile
index f869adf..5c50fb1 100644
--- NuttX/apps/examples/module/Makefile
+++ NuttX/apps/examples/module/Makefile
@@ -104,7 +104,9 @@ really_build: $(OBJS)
 	@touch .built
 
 .built:
+ifeq ($(CONFIG_EXAMPLES_MODULE),y)
 	@$(MAKE) -C drivers TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV)
+endif
 	@$(MAKE) TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" really_build
 
 ifeq ($(CONFIG_BUILD_KERNEL),y)
@@ -138,7 +140,9 @@ endif
 depend: .depend
 
 clean_drivers:
+ifeq ($(CONFIG_EXAMPLES_MODULE),y)
 	@$(MAKE) -C drivers TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)" CROSSDEV=$(CROSSDEV) clean
+endif
 
 clean: clean_drivers
 	$(call DELFILE, .built)
