--- a/stm32_ethernet.c
+++ b/stm32_ethernet.c
@@ -3974,14 +3974,25 @@
   ninfo("Reset the Ethernet block\n");
   stm32_ethreset(priv);
 
+#ifdef CONFIG_STM32F7_NO_PHY
+  //ignore phy-init in RMII DIRECT connection mode,
+  //config link manually as no auto-nego available;
+#ifdef CONFIG_STM32F7_ETHFD
+  priv->fduplex = 1;
+#endif
+#ifdef CONFIG_STM32F7_ETH100MBPS
+  priv->mbps100 = 1;
+#endif
+    
+#else
   /* Initialize the PHY */
-
   ninfo("Initialize the PHY\n");
   ret = stm32_phyinit(priv);
   if (ret < 0)
     {
       return ret;
     }
+#endif //CONFIG_STM32F7_NO_PHY
 
   /* Initialize the MAC and DMA */
 
