diff -u gcc-3.4.6/gcc/collect2.c.orig gcc-3.4.6/gcc/collect2.c
--- gcc-3.4.6/gcc/collect2.c.orig	2009-05-05 20:37:56.000000000 -0600
+++ gcc-3.4.6/gcc/collect2.c	2009-05-05 20:39:06.000000000 -0600
@@ -1534,7 +1534,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0666);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
