diff -u gcc-4.3.3/gcc/config.gcc.orig gcc-4.3.3/gcc/config.gcc
--- gcc-4.3.3/gcc/config.gcc.orig	2009-04-24 22:08:23.523688500 -0400
+++ gcc-4.3.3/gcc/config.gcc	2009-04-24 22:13:43.869488700 -0400
@@ -2880,9 +2880,9 @@
 
 		case "$with_arch" in
 		"" \
-		| armv[23456] | armv2a | armv3m | armv4t | armv5t \
-		| armv5te | armv6j |armv6k | armv6z | armv6zk \
-		| iwmmxt | ep9312)
+		| armv[234567] | armv2a | armv3m | armv4t | armv5t | armv5e \
+		| armv5te | armv6j |armv6k | armv6z | armv6zk | armv6t2 \
+		| armv7-a | armv7-r | armv7-m | ep9312 | iwmmxt)
 			# OK
 			;;
 		*)
